src/TensorNetwork.jl

```diff
@@ -261,20 +294,68 @@ function __check_index_sizes(tn)
     return true
 end
 
-const is_unsafe_region = ScopedValue(false) # global ScopedValue for the unsafe region
+struct UnsafeContext
+    refs::Vector{WeakRef}  # List of weak references
+
+    UnsafeContext() = new(Vector{WeakRef}())
+end
+
+Base.values(uc::UnsafeContext) = map(x -> x.value, uc.refs)
+
+# Global stack to manage nested unsafe contexts
+const _unsafe_context_stack = Ref{Vector{UnsafeContext}}(Vector{UnsafeContext}())
```

Comment:
mmm you don't need this? the idea was that `@unsafe_region` first creates a `UnsafeContext` and passes that the `TensorNetwork`

Reply:
You need this to check if there is an active `UnsafeContext` when you call `copy` and related functions.

Reply:
You need this to check if there is an active `UnsafeContext` when you call copy and related functions.